control-plane/plugin/src/resources/pool.rs

```diff
@@ -279,3 +330,81 @@ impl Label for Pool {
         Ok(())
     }
 }
+
+/// The PoolDisplayLabels structure is responsible for controlling the display formatting of Pool
+/// objects. `#[serde(flatten)]` and `#[serde(skip)]` attributes are used to ensure that when the
+/// object is serialised, only the `inner` object is represented.
+#[derive(Serialize, Debug)]
+pub struct PoolDisplayLabels {
```

Comment:
This is not just displaying the labels.
Also in the future we might add other info.

```suggestion
/// The PoolDisplay structure is responsible for controlling the display formatting of Pool
/// objects. `#[serde(flatten)]` and `#[serde(skip)]` attributes are used to ensure that when the
/// object is serialised, only the `inner` object is represented.
#[derive(Serialize, Debug)]
pub struct PoolDisplay {
```